button-card.js

```diff
@@ -41,22 +44,29 @@ class ButtonCard extends LitElement {
     const color = state.attributes.rgb_color ? `rgb(${state.attributes.rgb_color.join(',')})` : (
       config.color ? config.color : "var(--primary-text-color)"
     )
-    const color_on = state.state === 'on' ? color : "var(--disabled-text-color)";
+    
+    let color_custom='';
     let card_style = '';
     if (config.style){
       config.style.forEach( cssObject => {
         const attribute = Object.keys(cssObject)[0]
         const value = cssObject[attribute]
+        
+        if (attribute=="color") color_custom=value;
+
         card_style += `${attribute}: ${value};\n`
       })
     }
-    
+    const color_on = state.state === 'on' ? color : ( color_custom!==''? color_custom : "var(--disabled-text-color)"); 
```

Comment:
I'm a little bit confused by that. Why didn't you used the `color` setting in the config?

Is that because it get's overridden by the light attribute?

Reply:
Because you use this setting only when toggle is active, when it's not active you used only --disabled-text-color like icon's color.